src/sbl_filing_api/routers/filing.py

```diff
@@ -111,23 +113,32 @@ async def get_submission(request: Request, id: int):
 @router.put("/institutions/{lei}/filings/{period_code}/submissions/{id}/accept", response_model=SubmissionDTO)
 @requires("authenticated")
 async def accept_submission(request: Request, id: int, lei: str, period_code: str):
-    result = await repo.get_submission(request.state.db_session, id)
-    if not result:
+    submission = await repo.get_submission(request.state.db_session, id)
+    if not submission:
         return JSONResponse(
             status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
             content=f"Submission ID {id} does not exist, cannot accept a non-existing submission.",
         )
     if (
-        result.state != SubmissionState.VALIDATION_SUCCESSFUL
-        and result.state != SubmissionState.VALIDATION_WITH_WARNINGS
+        submission.state != SubmissionState.VALIDATION_SUCCESSFUL
+        and submission.state != SubmissionState.VALIDATION_WITH_WARNINGS
     ):
         return JSONResponse(
             status_code=status.HTTP_403_FORBIDDEN,
             content=f"Submission {id} for LEI {lei} in filing period {period_code} is not in an acceptable state.  Submissions must be validated successfully or with only warnings to be signed",
         )
-    result.state = SubmissionState.SUBMISSION_ACCEPTED
-    result.accepter = request.user.id
-    return await repo.update_submission(result, request.state.db_session)
+
+    await repo.update_submission_accepter(
+        request.state.db_session,
+        submission_id=id,
+        accepter=request.user.id,
+        accepter_name=request.user.name,
+        accepter_email=request.user.email,
+    )
+
+    submission.state = SubmissionState.SUBMISSION_ACCEPTED
+    submission = await repo.update_submission(submission, request.state.db_session)
```

Comment:
One thing I discovered working on the "/sign" endpoint is after the SubmissionAcceptorDAO is created in the db, you then have to assign that object to the Submission prior to the update.  Otherwise, what happens is the Submission object going to the update_submission *doesn't* have the acceptor field populated so the database will actually remove the relationship!  For instance, in the /sign I do this:

```
    filing = await repo.get_filing(request.state.db_session, lei, period_code)
    if not filing:
        return JSONResponse(
            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
            content=f"There is no Filing for LEI {lei} in period {period_code}, unable to sign a non-existent Filing.",
        )
    latest_sub = await repo.get_latest_submission(request.state.db_session, lei, period_code)
    if not latest_sub or latest_sub.state != SubmissionState.SUBMISSION_ACCEPTED:
        return JSONResponse(
            status_code=status.HTTP_403_FORBIDDEN,
            content=f"Cannot sign filing. Filing for {lei} for period {period_code} does not have a latest submission the SUBMISSION_ACCEPTED state.",
        )
    if not filing.contact_info:
        return JSONResponse(
            status_code=status.HTTP_403_FORBIDDEN,
            content=f"Cannot sign filing. Filing for {lei} for period {period_code} does not have contact info defined.",
        )
    sig = await repo.add_signature(request.state.db_session, filing_id=filing.id, user=request.user)
    filing.confirmation_id = lei + "-" + period_code + "-" + str(latest_sub.id) + "-" + str(sig.signed_date.timestamp())
    filing.signatures.append(sig)
    return await repo.upsert_filing(request.state.db_session, filing)
```
You don't have to append because you won't have a list of acceptors, but you still will want to do something similar.